crates/api_common/src/person.rs

```diff
@@ -70,6 +71,8 @@ pub struct SaveUserSettings {
   pub show_read_posts: Option<bool>,
   pub show_new_post_notifs: Option<bool>,
   pub discussion_languages: Option<Vec<LanguageId>>,
+  /// None leaves it as is, true will generate or regenerate it, false clears it out
+  pub generate_totp: Option<bool>,
```

Comment:
What does totp mean? Dont see that mentioned anywhere. At least for the API it should be clearer.

Reply:
Should I add more to the comment on this line, explaining what totp is?

Reply:
How about we rename `totp` to `2fa_secret` everywhere? Makes it much clearer what its for.

Reply:
2fa is a more general term that refers to both totp and hotp passwords. I'm only adding totp support, and don't plan on doing hotp ones (one time use passwords used for sms and email )

In the front end, I use the term 2fa just because its more well known.

Reply:
If hotp is not supported then i dont think there will be any confusion.

Reply:
I still would rather keep this totp, because its more technically accurate.

Reply:
Maybe accurate, but mainly confusing for anyone who isnt familiar with the term. You could add a rustdoc comment to every variable with this name, but that wouldnt help for the API. How about `2fa_secret_totp`?

Reply:
This one needs to have generate in the name, but how about I change everywhere I currently have `totp` to `2fa_totp` ?

Reply:
Yes thats okay.